specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,764 @@
+        HRESULT remove_CustomItemSelected(
+            [in] EventRegistrationToken token);
+    }
+
+    /// Represents a collection of `ContextMenuItem` objects. Used to get, remove and add
+    /// `ContextMenuItem` objects at the specified index.
+    [uuid(f562a2f5-c415-45cf-b909-d4b7c1e276d3), object, pointer_default(unique)]
+    interface ICoreWebView2ContextMenuItemCollection : IUnknown
+    {
+        /// Gets the number of `ContextMenuItem` objects contained in the `ContextMenuItemCollection`.
+        [propget] HRESULT Count([out, retval] UINT32* value);
+
+        /// Gets the `ContextMenuItem` at the specified index.
+        HRESULT GetValueAtIndex([in] UINT32 index,
+            [out, retval] ICoreWebView2ContextMenuItem** value);
+
+        /// Removes the `ContextMenuItem` at the specified index.
+        HRESULT RemoveValueAtIndex([in] UINT32 index);
+
+        /// Inserts the `ContextMenuItem` at the specified index.
+        HRESULT AddValueAtIndex([in] UINT32 index, [in] ICoreWebView2ContextMenuItem* value);
+    }
+
+    [uuid(76eceacb-0462-4d94-ac83-423a6793775e), object, pointer_default(unique)]
+    interface ICoreWebView2_4 : ICoreWebView2_3
+    {
+        /// Add an event handler for the `ContextMenuRequested` event.
+        /// `ContextMenuRequested` event is raised when a context menu is requested by the user
+        /// and the content inside WebView hasn't disabled context menus.
+        /// The host has the option to create their own context menu with the information provided in 
+        /// the event or can add items to or remove items from WebView context menu.
+        /// If the host doesn't handle the event, WebView will display the default context menu.
+        HRESULT add_ContextMenuRequested(
+            [in] ICoreWebView2StagingContextMenuRequestedEventHandler* eventHandler,
+            [out] EventRegistrationToken* token);
+
+        /// Remove an event handler previously added with `add_ContextMenuRequested`.
+        HRESULT remove_ContextMenuRequested(
+            [in] EventRegistrationToken token);
+    }
+
+    /// A continuation of the ICoreWebView2Environment interface.
+    [uuid(04d4fe1d-ab87-42fb-a898-da241d35b63c), object, pointer_default(unique)]
+    interface ICoreWebView2Environment : IUnknown
+    {
+        /// Create a `ContextMenuItem` object to insert into the WebView context menu.
+        /// The `IsChecked` property will only be used if the menu item kind is Radio or Checkbox.
+        /// For more information regarding paramters, see `ContextMenuItem`.
+        HRESULT CreateContextMenuItem(
+            [in] LPCWSTR label,
+            [in] LPCWSTR shortcutKeyDescription,
+            [in] IStream* iconStream,
+            [in] COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND kind,
+            [in] BOOL isEnabled,
+            [in] BOOL isChecked,
+            [out, retval] ICoreWebView2ContextMenuItem** item);
+    }
```

Comment:
`/// Create a `ContextMenuItemCollection` object.
HRESULT CreateContextMenuItemCollection(
[out, retval] ICoreWebView2ContextMenuItemCollection** collection);`
Needs to be added to allow developers to create Submenu items

Reply:
Per previous discussion, instead of adding a Create method, we can have the Children property be set to a valid and initialized collection and the end dev interacts with that collection rather than creating their own collection object.